src/libopenrave/robotconnectedbody.cpp

```diff
@@ -334,7 +334,7 @@ void RobotBase::ConnectedBody::GetResolvedLinks(std::vector<KinBody::LinkPtr>& l
     RobotBasePtr pattachedrobot = _pattachedrobot.lock();
     if( !!pattachedrobot ) {
         for(size_t ilink = 0; ilink < _vResolvedLinkNames.size(); ++ilink) {
-            links[ilink] = pattachedrobot->GetLink(_vResolvedLinkNames[ilink].first);
+            links.at(ilink) = pattachedrobot->GetLink(_vResolvedLinkNames.at(ilink).first);
```

Comment:
@cielavenir Since we're iterating with `vResolvedLinkNames.size()`, then `_vResolvedLinkNames[ilink]` should always be valid, and there's no reason to use `at`. Can you clean up this MR? thanks